include/tudat/simulation/estimation_setup/observationOutputSettings.h

```diff
@@ -32,22 +34,117 @@ enum ObservationDependentVariables
+        else if ( originatingLinkEndId_ != LinkEndId( "", "" ) )
+        {
+            identifier += " to link end: (" + originatingLinkEndId_.bodyName_ + ", " + originatingLinkEndId_.stationName_ + ")";
+        }
+        else if ( originatingLinkEndType_ != unidentified_link_end )
+        {
+            identifier += " to link end of type " + getLinkEndTypeString( originatingLinkEndType_ );
+        }
+        return identifier;
     }
 
-    virtual ~ObservationDependentVariableSettings( ){ }
+    virtual std::string getIdentifier( )
+    {
+        return getBaseIdentifier( );
+    }
 
-    ObservationDependentVariables variableType_;
+    virtual bool areSettingsCompatible( const std::shared_ptr< ObservationDependentVariableSettings > otherSettings )
+    {
+        return areBaseSettingsCompatible( otherSettings, true );
+    }
+
+    bool areBaseSettingsCompatible( const std::shared_ptr< ObservationDependentVariableSettings > otherSettings,
+                                    const bool revertedLinksAllowed = true )
+    {
+        bool isCompatible = true;
+        if ( variableType_ != otherSettings->variableType_ )
+        {
+            isCompatible = false;
+        }
+        else
+        {
+            bool directLinksMatch =
+                    ( ( linkEndType_ == otherSettings->linkEndType_ || linkEndType_ == unidentified_link_end || otherSettings->linkEndType_ == unidentified_link_end )
+                    && ( ( linkEndId_ == otherSettings->linkEndId_ || linkEndId_ == LinkEndId( "", "" ) || otherSettings->linkEndId_ == LinkEndId( "", "" ) ) )
+                    && ( ( originatingLinkEndType_ == otherSettings->originatingLinkEndType_ ||
+                    originatingLinkEndType_ == unidentified_link_end || otherSettings->originatingLinkEndType_ == unidentified_link_end ) )
+                    && ( ( originatingLinkEndId_ == otherSettings->originatingLinkEndId_ ||
+                    originatingLinkEndId_ == LinkEndId( "", "" ) || otherSettings->originatingLinkEndId_ == LinkEndId( "", "" ) ) ) );
+
+            bool revertedLinksMatch =
+                    ( ( linkEndType_ == otherSettings->originatingLinkEndType_ || linkEndType_ == unidentified_link_end || otherSettings->originatingLinkEndType_ == unidentified_link_end )
+                    && ( ( linkEndId_ == otherSettings->originatingLinkEndId_ || linkEndId_ == LinkEndId( "", "" ) || otherSettings->originatingLinkEndId_ == LinkEndId( "", "" ) ) )
+                    && ( ( originatingLinkEndType_ == otherSettings->linkEndType_ ||
+                    originatingLinkEndType_ == unidentified_link_end || otherSettings->linkEndType_ == unidentified_link_end ) )
+                    && ( ( originatingLinkEndId_ == otherSettings->linkEndId_ ||
+                    originatingLinkEndId_ == LinkEndId( "", "" ) || otherSettings->linkEndId_ == LinkEndId( "", "" ) ) ) );
+
+            if ( !directLinksMatch && ( !revertedLinksAllowed || !revertedLinksMatch ) )
```

Comment:
This is a cool way to check whether the order of the links is relevant (right?)

Reply:
yes exactly, to give more flexibility regarding the input arguments order when defining the variable settings  (for some dependent variables, it doesn't matter if the user gets the link end order wrong). This also prevents the dependent variables being created and computed twice if the link "direction" does not matter.